packages/zosuss/src/SshBaseHandler.ts

```diff
@@ -58,22 +59,108 @@ export abstract class SshBaseHandler implements ICommandHandler {
+            {
+                propertyName: "privateKey",
+                propertiesOverridden: [
+                    "password",
+                    "tokenType",
+                    "tokenValue",
+                    "cert",
+                    "certKey",
+                ],
+            },
+        ];
+        const sshSessCfg: ISshSession = SshSession.createSshSessCfgFromArgs(
+            commandParameters.arguments
         );
+        let sshSessCfgWithCreds =
+            await ConnectionPropsForSessCfg.addPropsOrPrompt<ISshSession>(
+                sshSessCfg,
+                commandParameters.arguments,
+                {
+                    parms: commandParameters,
+                    propertyOverrides: sshSessCfgOverride,
+                    supportedAuthTypes: [SessConstants.AUTH_TYPE_BASIC],
+                }
+            );
         this.mSession = new SshSession(sshSessCfgWithCreds);
 
         this.mArguments = commandParameters.arguments;
-        await this.processCmd(commandParameters);
+
+        try {
+            await this.processCmd(commandParameters);
+        } catch (e) {
+            this.console.log("Initial key passphrase authentication failed!" + "\n");
+            if (
+                e.message.includes("but no passphrase given") ||
+                e.message.includes("bad passphrase?")
+            ) {
+                const maxAttempts = 3;
+                let attempt = 0;
+                let success = false;
+                while (attempt < maxAttempts && !success) {
+                    try {
+                        sshSessCfgWithCreds =
+                            await ConnectionPropsForSessCfg.addPropsOrPrompt<ISshSession>(
+                                sshSessCfgWithCreds,
+                                commandParameters.arguments,
+                                {
+                                    parms: commandParameters,
+                                    propertyOverrides: sshSessCfgOverride,
+                                    supportedAuthTypes: [
+                                        SessConstants.AUTH_TYPE_BASIC,
+                                    ],
+                                    propsToPromptFor: [
+                                        {
+                                            name: "keyPassphrase",
+                                            isGivenValueValid: (givenValue: {
+                                                [key: string]: any;
+                                            }) => {
+                                                let saveKP: boolean = true;
+                                                const result = utils.parseKey(
+                                                    fs.readFileSync(
+                                                        sshSessCfgWithCreds[
+                                                            "privateKey"
+                                                        ]
```

Comment:
Please use dot notation for referencing properties when possible 😋 
```suggestion
                                                        sshSessCfgWithCreds.privateKey
```